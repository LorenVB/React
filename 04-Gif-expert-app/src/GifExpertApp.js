import React, { Fragment, useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState([]);

    return (<Fragment>
        <h1>GifExpertApp</h1>
        <AddCategory setCategories={setCategories} />
        <hr />

        {/* <button onClick={handleAdd}>Agregar</button> */}
        <ol>
            {
                categories.map((category) => (
                    <GifGrid
                        key={category}
                        category={category} />
                )
                )}
        </ol>
    </Fragment>
    )
}
