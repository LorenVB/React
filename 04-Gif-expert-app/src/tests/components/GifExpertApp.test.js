import React from 'react';
import {shallow} from 'enzyme'
import { GifExpertApp } from "../../GifExpertApp"

describe('Testing GifExpertApp component', () => { 
    test('should match with the snapshot', () => { 
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot()
     })
 })