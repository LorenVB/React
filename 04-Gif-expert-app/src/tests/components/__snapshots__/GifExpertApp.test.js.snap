// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp component should match with the snapshot 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
